### PR TITLE
Better look for steps

### PR DESCRIPTION
**Pull request by DavertMik**
Selection_585

Selection_586

substeps

This PR includes

###  Unification of steps. 

Motivation: we can't hardcode step names in the components. There are different helpers, different steps names, users can create their own helpers and step names. The only rule we have that there are next groups of tests:

* actions
* assertions
* waiters
* grabbers
* comments
* metasteps

Hardcoding look for the exact step will generate tons of issues for edge cases.

### Unified Arguments

This can be doubtable change, but because arguments can be long, short, and there are up to 3 argument, there should be a way to display them nicely no matter of an actual step being used.

In this approach,

 I use inline style for an argument and a color to show its position

* blue -> 1st argument
* orange -> 2nd
* green -> 3rd

This is discussible, but it brings a clean look to arguments.

### Nested MetaSteps

The hierarchical structure of a test is now shown in a report.
All actions from a metastep will be shown with an indentation.
Clicking on metastep will toggle actions. So for a long scenarios you can easily toggle them up and down.

### Using Tailwind CSS

Tailwind CSS provides sane base classes which can be combined to create unified look for components. Such as steps.

### CODE REVIEW

**Comment by DavertMik** [comment]
Updated for grabbers, locators, and secrets
Selection_587

**Comment by DavertMik** [comment]
Once approved I will fix all linting issues we have now (mostly because of non-used components)

**Comment by DavertMik** [comment]
Fixing flaws in current design:

Selection_588

File: package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
     "moment": "^2.24.0",
     "sass": "^1.22.9",
     "sass-loader": "^7.1.0",
+    "tailwindcss": "^1.1.2",
```
Comment (hubidu) [inline package.json]: You sure you wanna use this? Another css library. What about buefiy/bulma? Do you want to remove them?

Comment (DavertMik) [reply package.json]: No, they are different. Bulma looks nice as a component library. 
Tailwind is a pack of styles to create own components. 

So I can just set for a block: 

* create nice shadows
* make nice red background 

etc. I think I can reduce impact by removing some classes out of it, but I really liked how easily I can create custom components with it. I hope they can live together with bulma

Comment (DavertMik) [reply package.json]: I think you compare Tailwind CSS to https://bulma.io/documentation/modifiers/
But there are much more styles in Tailwind than in Bulma

File: src/components/Step.vue [hunk 2]
```diff
@@ -248,24 +313,73 @@ export default {
 }
 </script>
 
-<style lang="scss" scoped>
+
+<style lang="scss">
```
Comment (hubidu) [inline src/components/Step.vue]: We should decide how we want to write CSS. Should we use SASS or not. Naming conventions for CSS classes (like SuitCSS), use Tailwind or Bulma etc.
Otherwise everything will soon be a total mess.

Comment (DavertMik) [reply src/components/Step.vue]: I agree. 
I vote for sass 🙂

File: package.json (hunk 1 above)
Comment (DavertMik) [reply package.json]: From Tailwind website: 

> Unlike many other CSS frameworks, Tailwind doesn't include any component classes like form-input, btn, card, or navbar.

File: src/components/steps/AssertionStep.vue [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="step assert">
+
+    <i class="far fa-eye" v-if="!isNegative(step.name)"></i>
```
Comment (hubidu) [inline src/components/steps/AssertionStep.vue]: Why did you bring the icons back? I thought you said you find them unintuitive.

Comment (DavertMik) [reply src/components/steps/AssertionStep.vue]: Ha-ha, I took a second round to think 😃
Well, if there is a common principle, why not to use icons if they save space and focus.

So I decided to use icons for waiters, grabbers, assertions.
As well as custom styles for those step types.

Comment (DavertMik) [reply src/components/steps/AssertionStep.vue]: I think my initial dislike was based on the fact that each step had its own icon. Which was not very clear - which one have what icon and on which principle they are applied

File: package.json (hunk 1 above)
Comment (hubidu) [reply package.json]: I know the concept behind tailwind. I still don't think it's a good idea to have both in the same project.

Comment (DavertMik) [reply package.json]: They do not intersect, really. I would keep both. 

* Bulma for standard controls/grid/etc
* Tailwind for constructing custom components

I think I can turn off initial Tailwind classes so they won't pollute global CSS namespace, and use them only in styles (as I do it via `@apply`)

**Comment by DavertMik** [comment]
No objections/conversations for a month. Merging...
